diff --git a/node_modules/@astrojs/image/dist/vite-plugin-astro-image.js b/node_modules/@astrojs/image/dist/vite-plugin-astro-image.js
index 5f199c8..7887239 100644
--- a/node_modules/@astrojs/image/dist/vite-plugin-astro-image.js
+++ b/node_modules/@astrojs/image/dist/vite-plugin-astro-image.js
@@ -39,8 +39,7 @@ function createPlugin(config, options) {
         });
         meta.src = `__ASTRO_IMAGE_ASSET__${handle}__`;
       } else {
-        const relId = path.relative(fileURLToPath(config.srcDir), id);
-        meta.src = join("/@astroimage", relId);
+        meta.src = "/@astroimage/"  + url.pathname;
         meta.src = slash(meta.src);
       }
       return `export default ${JSON.stringify(meta)}`;
@@ -50,7 +49,7 @@ function createPlugin(config, options) {
         var _a;
         if ((_a = req.url) == null ? void 0 : _a.startsWith("/@astroimage/")) {
           const [, id] = req.url.split("/@astroimage/");
-          const url = new URL(id, config.srcDir);
+          const url = new URL(id, 'file:');
           const file = await fs.readFile(url);
           const meta = await metadata(url);
           if (!meta) {
